cmd/acme/cert.go

```diff
@@ -215,11 +220,27 @@ func pollCert(url string) [][]byte {
 	}
 }
 
-func challengeFile(domain, content string) (string, error) {
-	f, err := ioutil.TempFile("", domain)
+func challengeFile(challengeFilename, domain, content string) (string, error) {
+	// If ACME_CHALLENGE_DIR is set, place the challenge into it.
+	// Otherwise, create a temporary file
+	var f *os.File
+	var err error
+	if os.Getenv("ACME_CHALLENGE_DIR") != "" {
+		// Create .well-known directory
+		challengeDir := filepath.Join(os.Getenv("ACME_CHALLENGE_DIR"), ".well-known/acme-challenge")
```

Comment:
I thought it was more convenient to automatically append `.well-known/challenge`. The protocol requires this exact path and having to specify it manually seems error prone.

Not sure what you mean though: Do you mean it's a good thing to specify the ful path manually, or would you rather keep it automatic?